drivers/memc/memc_mspi_aps6404l.c

```diff
@@ -341,28 +343,34 @@ static int memc_mspi_aps6404l_init(const struct device *psram)
 	}
 	data->dev_cfg = cfg->tar_dev_cfg;
 
+#if CONFIG_MSPI_TIMING
 	if (mspi_timing_config(cfg->bus, &cfg->dev_id, cfg->timing_cfg_mask,
 			       (void *)&cfg->tar_timing_cfg)) {
 		LOG_ERR("Failed to config mspi timing/%u", __LINE__);
 		return -EIO;
 	}
 	data->timing_cfg = cfg->tar_timing_cfg;
+#endif /* CONFIG_MSPI_TIMING */
 
+#if CONFIG_MSPI_XIP
```

Comment:
Why are these ifdef's included of the enable is already there?

To save code space?

Reply:
When XIP is not supported by a specific hardware, CONFIG_MSPI_XIP=n is used to indicate that. This also means that xip_config may not be implemented, so calling it in device driver will result in error. Leaving it out of compile will not only save code space but also ensure the device driver is still functional without optional features.